lib/Frontend/CompilerInvocation.cpp

```diff
@@ -47,6 +47,22 @@ static void updateRuntimeLibraryPath(SearchPathOptions &SearchPathOpts,
   llvm::sys::path::append(LibPath, getPlatformNameForTriple(Triple));
   SearchPathOpts.RuntimeLibraryPath = LibPath.str();
 
+  // armv6l and armv7l (which come from linux) are mapped to armv6 and 
+  // armv7 (respectively) within llvm.  When a Triple is created by llvm,
+  // the string is preserved, which keeps the 'l'.  The libraries are built
+  // without this extra character, so we need to strip it out when using
+  // the ArchName in the RuntimeLibraryPath. 
+  // By explicitly setting the architecture to the subtype that it aliases to,
+  // we remove that extra character while not introducing other side effects.
+  if (Triple.getOS() == llvm::Triple::Linux) {
```

Comment:
I'm a little concerned about this function mutating `Triple`. It should probably be passed const.

LLVM does have logic to canonicalize a triple. If you do that early on in the Driver, does everything work?